Apigee/SmartDocs/Revision.php

```diff
@@ -401,12 +409,28 @@ public function delete($revisionId = null)
      */
     public function importSwagger($swaggerUrl)
     {
+        $combinedString = rawurlencode($this->apiId) . '/revisions?action=import&format=swagger';
         $this->blankValues();
-        $this->post('?action=import&format=swagger', 'URL=' . $swaggerUrl, 'text/plain; charset=utf-8');
+        $this->post($combinedString, 'URL=' . $swaggerUrl, 'text/plain; charset=utf-8');
         $response = $this->responseObj;
         self::fromArray($this, $response);
     }
 
+  /**
+   * Imports a model revision from a WADL document.
+   *
+   * @param string $modelId
+   * @param string $xml
+   */
+  public function importSwaggerFile($yaml)
```

Comment:
This function does not match the API.   You need to match the API by deprecating the old methods, then creating two new methods:
1. importFile()
2. importURL()

What happened to the method signature I gave you? Make sure to add the notes I put in this PHP Doc.
Here it is again:

```php
/**
     * Import a model from a file.
     *
     * The file is passed into the method as a string. Note that you do not have
     * to create a revision first, this method will automagically create the
     * revision for you.
     *
     * Note that Swagger 1.2 cannot be expressed as in a single-file format, so
     * importFile can only be used with Swagger 2.0.  For Swagger 1.2, use
     * importUrl() instead.
     *
     * @param string $document The text to import into the model.
     * @param string $document_format The format, either 'wadl', 'swagger',
     * or 'apimodel'.
     * @param null|string $modelId The model id, if not passed will be the modelId
     * from this object.
     *
     * @throws ParameterException
     */
    public function importFile($document, $document_format, $modelId = null)
    {
        $modelId = $modelId ?: $this->id;
        if (empty($modelId)) {
            throw new ParameterException('Cannot load a model with no ID.');
        }
        $this->blankValues();
        // TODO: change 'application/xml if it is Swagger.
        $this->post('?action=import&format=wadl', $xml, 'application/xml; charset=utf-8');
        $response = $this->responseObj;
        self::fromArray($this, $response);
    }
```